src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -447,6 +453,24 @@ internal string AdvanceThroughBalancedParentheticalExpression()
             return expressionText;
         }
 
+        internal void TakeTextPositionSnapshot()

```

Comment:
Ya, my thought was that taking a snapshot in progress would not take a new snapshot.